### DISCUSSION TITLE
HA-device send msg.payload value

### DISCUSSION THREAD

**Discussion by hjalmarnilsson** [zachowj/node-red-contrib-home-assistant-websocket#777] [Q&A] [answered]
Hi,
I'm using the HA Device node to set a value down to a Device connected to my Home assistant.

My config is this:
Type: Action
Device Nibe heatpump intergation
Trigger Set Value for Speed Circ-pump Heat
Value 4
This all works fine, but I would like to send the payload of the incoming msg as the value instead of a fixed value.
I've tried to mustash it, but as you by now understand I'm a complete noob so I can't for my life get this to work.
How do I set the incoming msg.payload as the value sent by to this integration?
Please help and point out my stupidity and explain how I write in the value box.
I've tried.
msg.payload
payload
{{payload}}
but all that happens is that the value is empty after I click it again after depoloying it

**Comment by zachowj** [comment, marked as answer]
Currently, it is not possible to use dynamic values in the device node.

**Reply by hjalmarnilsson** [reply]
Ok thanks for quick. response!
Can you think of any way I could get around that or shall I forget about it for the next frew months? 
Not being an arse, just need to know if I should code this in a different way to get around the problem.

**Reply by zachowj** [reply]
There's no simple way but it is possible. If there's an expose service that you can use via the call-service node that would be the easiest. If not the below example should work.

Build an action in the HA automation section using the UI. Then edit it YAML so you can copy the code. Using an online YAML to JSON converter change the YAML to JSON.

Then in NR take that converted JSON and use an API node to send that action to HA.

The format of the API is 
```
{
  "type": "nodered/device_action",
  "action": <CONVERTED JSON variable substitutions can also be placed here using JSONata>
}
```

```
[{"id":"e2d23f7a282937bd","type":"ha-api","z":"c89d915bdff0f798","name":"","server":"","version":1,"debugenabled":false,"protocol":"websocket","method":"get","path":"","data":"{\t   \"type\": \"nodered/device_action\",\t   \"action\": {\t       \"device_id\": \"aeb98827e\",\t       \"domain\": \"zwave_js\",\t       \"type\": \"set_config_parameter\",\t       \"parameter\": payload,\t       \"bitmask\": 1,\t       \"subtype\": \"7[0x1] (Send Status Change Report: Local Control)\"\t    }\t};","dataType":"jsonata","responseType":"json","outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"results"}],"x":798,"y":112,"wires":[[]]},{"id":"cffc3ddaf3e5b050","type":"inject","z":"c89d915bdff0f798","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":444,"y":112,"wires":[["4e38f32bd35fc9c9"]]},{"id":"4e38f32bd35fc9c9","type":"change","z":"c89d915bdff0f798","name":"get custom value","rules":[{"t":"set","p":"payload","pt":"msg","to":"5","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":618,"y":112,"wires":[["e2d23f7a282937bd"]]}]
```

**Reply by hjalmarnilsson** [reply]
That worked! Thanks!
